pkg/hubble/relay/observer/server.go

```diff
@@ -356,3 +375,44 @@ func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusR
 	}
 	return resp, g.Wait()
 }
+
+func (s *Server) heartbeat() {
+	ticker := time.NewTicker(heartbeatInterval)
+	done := make(chan bool)
+
+	s.heartbeatTicker = ticker
+	s.heartbeatDone = done
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				ticker.Stop()
+				return
+			case <-ticker.C:
+				status := s.peerConnectivityStatus()
+				s.metrics.ObserveNodesStatus(status)
+			}
+		}
+	}()
+}
+
+func (s *Server) peerConnectivityStatus() map[relaypb.NodeState]uint32 {
+	var numConnectedPeers, numUnavailablePeers uint32
+	peers := s.peers.List()
+	for _, p := range peers {
+		if !isAvailable(p.Conn) {
```

Comment:
the `isAvailable` categorizes grpc connection (between relay and cilium-agent) states into 2 categories. The "Ready" and "Idle" states are  categorized as available. All other states are categorized as unavailable.

When debugging peer connectivity issues I believe it can be more useful to report the actual grpc connection state as defined in https://github.com/cilium/cilium/blob/main/vendor/google.golang.org/grpc/connectivity/connectivity.go#L31